### PR TITLE
Tl dsdegp 1481 uuids

### PR DESCRIPTION
**Pull request by tlangs**
### Description

Made Clio assign V1 UUIDs to every upsert.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: project/src/main/scala/org/broadinstitute/clio/sbt/Dependencies.scala [hunk 1]
```diff
@@ -139,4 +141,8 @@ object Dependencies {
     "org.scalatest" %% "scalatest" % ScalaTestVersion,
     "org.slf4j" % "slf4j-api" % Slf4jVersion
   ).map(_ % IntegrationTest)
+
+  val ThirdPartyDependencies: Seq[MavenRepository] = Seq(
```
Comment (danxmoran) [inline project/src/main/scala/org/broadinstitute/clio/sbt/Dependencies.scala]: Naming nitpick: `ThirdPartyResolvers` would be better.

**Review by danxmoran — COMMENTED** [review]
Existing code looks good. One request and one question:
1. Can you add a new integration test that adds a bunch of ubams and asserts their returned IDs are all unique?
2. Instead of returning the raw Long, would it be better to make a new transfer model for add-outputs, so the returned number can have a label?

**Comment by tlangs** [comment]
@danxmoran I like the transfer model idea. I just didn't know if it was worth it for just one Long.

File: project/src/main/scala/org/broadinstitute/clio/sbt/Dependencies.scala [hunk 2]
```diff
@@ -49,7 +50,8 @@ object Dependencies {
     "io.circe" %% "circe-generic" % CirceVersion,
     "io.circe" %% "circe-generic-extras" % CirceVersion,
     "io.circe" %% "circe-parser" % CirceVersion,
-    "org.slf4j" % "slf4j-api" % Slf4jVersion
+    "org.slf4j" % "slf4j-api" % Slf4jVersion,
+    "com.eaio.uuid" % "uuid" % UUIDVersion
```
Comment (tbl3rd) [inline project/src/main/scala/org/broadinstitute/clio/sbt/Dependencies.scala:54]: It looks like this shells out to '/sbin/ifconfig -a' for the MAC address,
which is ... not cool.  See .../com/eaio/uuid/UUIDGen.java

File: clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/DocumentWgsUbam.scala [hunk 3]
```diff
@@ -8,6 +8,7 @@ case class DocumentWgsUbam(flowcellBarcode: String,
                            lane: Int,
                            libraryName: String,
                            location: Location,
+                           clioId: Option[Long],
```
Comment (tbl3rd) [inline clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/DocumentWgsUbam.scala]: This should have a not_analyzed string field mapping.

File: clio-server/src/main/scala/org/broadinstitute/clio/server/service/WgsUbamService.scala [hunk 4]
```diff
@@ -18,19 +19,23 @@ class WgsUbamService private (
   def upsertMetadata(
     transferKey: TransferWgsUbamV1Key,
     transferMetadata: TransferWgsUbamV1Metadata
-  ): Future[Unit] = {
+  ): Future[Long] = {
+    val clioId = new UUID().getClockSeqAndNode
```
Comment (tbl3rd) [inline clio-server/src/main/scala/org/broadinstitute/clio/server/service/WgsUbamService.scala]: This is just the MAC and the seeded sequence number.

File: project/src/main/scala/org/broadinstitute/clio/sbt/Dependencies.scala (hunk 1 above)
Comment (tbl3rd) [inline project/src/main/scala/org/broadinstitute/clio/sbt/Dependencies.scala]: Ugh.

**Review by tbl3rd — CHANGES_REQUESTED** [review]
The plumbing looks OK, but we need to find another generator for clioId. Use the entire UUID clioId and map it as a not_analyzed string ... while ... we find a better ID implementation.

**Pull request closed without merging**